src/main/java/com/minecolonies/entity/ai/citizen/farmer/EntityAIWorkFarmer.java

```diff
@@ -1,647 +1,560 @@
+     * @param field the field close to this position.
+     * @return true if should be hoed.
+     */
+    private boolean shouldPlant(BlockPos position, Field field)
+    {
+        ItemStack itemStack = BlockUtils.getItemStackFromBlockState(world.getBlockState(position.up()));
 
-                    return true;
-                }
-            }
+        if(itemStack != null && itemStack.getItem() == field.getSeed())
+        {
+            requestSeeds = false;
         }
-        return false;
+        return !field.isNoPartOfField(world, position) && world.getBlockState(position.up()) instanceof IGrowable
+               && !(world.getBlockState(position).getBlock() instanceof BlockHutField) && world.getBlockState(position).getBlock() == Blocks.farmland;
     }
 
-    private void dumpInventory()
+    /**
+     * Checks if the ground should be hoed and the block above removed.
+     * @param position the position to check.
+     * @param field the field close to this position.
+     * @return true if should be hoed.
+     */
+    private boolean shouldHoe(BlockPos position, Field field)
+    {
+        return !field.isNoPartOfField(world, position) && !(world.getBlockState(position.up()).getBlock().getItem(world, position) instanceof ItemSeeds)
+               && !(world.getBlockState(position).getBlock() instanceof BlockHutField)
+               && (world.getBlockState(position).getBlock() == Blocks.dirt || world.getBlockState(position).getBlock() == Blocks.grass);
+    }
+
+    /**
+     * Searches and adds a field that has not been taken yet for the farmer and then adds it to the list.
+     */
+    private void searchAndAddFields()
     {
-        if (BlockPosUtil.isWorkerAtSiteWithMove(worker, worker.getWorkBuilding().getLocation()))
+        Colony colony = worker.getColony();
+        if(colony != null)
         {
-            for (int i = 0; i < getInventory().getSizeInventory(); i++)
+            Field newField = colony.getFreeField();
+
+            if(newField!=null && getOwnBuilding() != null)
             {
-                ItemStack stack = getInventory().getStackInSlot(i);
-                if (stack != null && !isStackTool(stack) && worker.getWorkBuilding().getTileEntity() != null)
-                {
-                    ItemStack returnStack = InventoryUtils.setStack(worker.getWorkBuilding().getTileEntity(),
-                            stack);
-                    if (returnStack == null)
-                    {
-                        getInventory().decrStackSize(i, stack.stackSize);
-                    }
-                    else
-                    {
-                        getInventory().decrStackSize(i, stack.stackSize - returnStack.stackSize);
-                    }
-                }
+                getOwnBuilding().addFarmerFields(newField);
             }
-            job.setStage(Stage.WORKING);
         }
     }
 
-    private static boolean isStackTool(ItemStack stack)
+    /**
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) First sentence should end with a period. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=checkstyle%3AJavadoc_Style)